Prog/clean-DC.Rmd

```diff
@@ -0,0 +1,1115 @@
+---
+title: "Regional Housing Framework"
+subtitle: "Clean DC Black Knight data"
+author: ""
+output:
+  html_document:
+    number_sections: FALSE
+    self_contained: TRUE
+    code_folding: show
+    toc: TRUE
+    toc_float: TRUE
+    css: ../www/web_report.css
+    editor_options:
+      chunk_output_type: console
+---
+
+<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato" />
+
+Library: RegHsg
+
+Project: Regional Housing Framework
+
+Author: Yipeng Su adapted from Sarah Strochak
+
+Version: R 3.5.1, RStudio 1.1.423
+
+Last updated `r format(Sys.time(), '%B %d, %Y')`
+
+Environment: Local Windows session (desktop)
+
+
+```{r rmarkdown-setup, echo = FALSE}
+knitr::opts_chunk$set(warning = FALSE)
+knitr::opts_chunk$set(message = FALSE)
+# knitr::opts_chunk$set(eval = FALSE)
+```
+
+## Description
+
+There are three main goals of the jurisdiction level cleaning process:
+
+1. Recategorize the county land use codes into more general codes
+
+2. Collapse observations at the address level
+
+3. Clean the variables needed to provide density estimates
+
+
+## Set-up
+#### Load libraries and functions
+```{r setup}
+library(tidyverse)
+library(DescTools)
+library(purrr)
+
+source("../Macros/read-bk.R")
+source("../Macros/filter-bk.R")
+source("../Macros/select-vars.R")
+source("../Macros/sample-properties.R")
+source("../Macros/classify-addresses.R")
+```
+#### Create directory for data exports on local computer
+```{r dir}
+if (!dir.exists("../Data")) {
+  dir.create("../Data")
+}
+
+```
+
+
+#### Set FIPS code and filepath name
+```{r fips}
+currentfips <- "11001"
+filepath <- "DC"
+jdir <- paste0("L:/Libraries/RegHsg/Data/", filepath, "/")
+```
+
+#### Load in precleaned Black Knight data for the region, select jurisdiction and standard variables
+```{r read}
+
+jur <- readRDS(paste0(jdir, 
+               "precleaned-",
+               filepath,
+               "-data.Rdata"))
+    
+```
+## Recatagorize county land use
+In order to complete the vacant land and soft site analysis, we will need to break down properties into different classifications. This process creates four variables:
+
+1. residential: 1 for residential, 0 for other.
+2. category: categories are sf, mf, commercial, office, and vacant.
+3. category_detail: this will vary by jurisdiction, but  includes the most detail possible to generalize from the county land use codes.
+4. building_type: this indicates the building type for multifamily parcels (condos and apartments)
+
+Export county land use codes for manual classification
+```{r export-landuse}
+currentjur_county <- jur %>% 
+  group_by(countylandusedescription) %>% 
```

Comment:
can you do a file that has the black knight land use vacant codes cross tabbed with the usecode from parcel_base?